spec.emu

```diff
@@ -21,110 +21,45 @@ contributors: Jordan Harband
 	</emu-alg>
 </emu-clause>
 
-<emu-clause id="sec-promisereaction-records">
-	<h1>PromiseReaction Records</h1>
-	<p>…</p>
-	<emu-table id="table-58" caption="PromiseReaction Record Fields">
-		<table>
-			<tbody>
-			<tr>
-				<th>
-					Field Name
-				</th>
-				<th>
-					Value
-				</th>
-				<th>
-					Meaning
-				</th>
-			</tr>
-			<tr>
-				<td>
-					…
-				</td>
-				<td>
-					…
-				</td>
-				<td>
-					…
-				</td>
-			</tr>
-			<tr>
-				<td>
-					[[Type]]
-				</td>
-				<td>
-					Either `"Fulfill"`, `"Reject"`, or `"Finally"`.
-				</td>
-				<td>
-				</td>
-			</tr>
-			<tr>
-				<td>
-					…
-				</td>
-				<td>
-					…
-				</td>
-				<td>
-					…
-				</td>
-			</tr>
-			</tbody>
-		</table>
-	</emu-table>
-</emu-clause>
-
 <emu-clause id="sec-performpromisefinally" aoid="PerformPromiseFinally">
 	<h1>PerformPromiseFinally ( _promise_, _onFinally_, _resultCapability_ )</h1>
 	<p>The abstract operation PerformPromiseFinally performs the &ldquo;finally&rdquo; operation on _promise_ using _onFinally_ as its settlement actions. The result is _resultCapability_'s promise.</p>
 	<emu-alg>
 		1. Assert: IsPromise(_promise_) is *true*.
 		1. Assert: _resultCapability_ is a PromiseCapability Record.
 		1. If IsCallable(_onFinally_) is *false*, then
-			1. Let _onFinally_ be *undefined*.
-		1. Let _reaction_ be the PromiseReaction { [[Capability]]: _resultCapability_, [[Type]]: `"Finally"`, [[Handler]]: _onFinally_ }.
-		1. Return EnqueuePromiseReactions(_promise_, _reaction_, _reaction_, _resultCapability_).
+			1. Return PerformPromiseThen(_promise_, *undefined*, *undefined*, _resultCapability_);
+		1. Let _thenFinally_ be ! CreateThenFinally(_onFinally_).
+		1. Let _catchFinally_ be ! CreateCatchFinally(_onFinally_).
+		1. Return PerformPromiseThen(_promise_, _thenFinally_, _catchFinally_, _resultCapability_).
 	</emu-alg>
 </emu-clause>
 
-<emu-clause id="sec-performpromisethen" aoid="PerformPromiseThen">
-	<h1>PerformPromiseThen ( _promise_, _onFulfilled_, _onRejected_, _resultCapability_ )</h1>
-	<p>The abstract operation PerformPromiseThen performs the &ldquo;then&rdquo; operation on _promise_ using _onFulfilled_ and _onRejected_ as its settlement actions. The result is _resultCapability_'s promise.</p>
+<emu-clause id="sec-createthenfinally" aoid="CreateThenFinally">
```

Comment:
This isn't the appropriate format for creating closures. See e.g. the Promise.all resolve element functions.